subsys/usb/usb_device.c

```diff
@@ -612,14 +657,35 @@ static bool usb_set_interface(u8_t iface, u8_t alt_setting)
 				if_desc = (void *)p;
 			}
 
-			LOG_DBG("iface_num %u alt_set %u", iface, alt_setting);
+			LOG_DBG("Current iface %u alt setting %u",
+				cur_iface, cur_alt_setting);
 			break;
 		case USB_ENDPOINT_DESC:
-			if ((cur_iface != iface) ||
+			if ((cur_iface == iface) &&
 			    (cur_alt_setting != alt_setting)) {
+				LOG_DBG("Disable endpoint for interface %d, "
+					"alt setting %d",
+					 cur_iface, cur_alt_setting);
+				ret_val = reset_endpoint((struct usb_ep_descriptor *)p);
+				if (!ret_val) {
```

Comment:
Better to log in reset_endpoint for both calls usb_dc_ep_configure and usb_dc_ep_disable, please remove then L670...L675, code here would also looks cleaner.